ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h

```diff
@@ -374,13 +385,27 @@ HydroMechanicsLocalAssembler<ShapeFunctionDisplacement, ShapeFunctionPressure,
         medium->property(MPL::PropertyType::reference_temperature)
             .template value<double>(vars, x_position, t, dt);
 
+    auto const& identity2 = MathLib::KelvinVector::Invariants<
+        MathLib::KelvinVector::KelvinVectorDimensions<DisplacementDim>::value>::
+        identity2;
+
     for (unsigned ip = 0; ip < n_integration_points; ip++)
     {
         x_position.setIntegrationPoint(ip);
 
-        vars[static_cast<int>(MPL::Variable::phase_pressure)] =
-            _ip_data[ip].N_p.dot(p);
+        double const p_int_pt = _ip_data[ip].N_p.dot(p);
+        vars[static_cast<int>(MPL::Variable::phase_pressure)] = p_int_pt;
 
+        auto const alpha = solid.property(MPL::PropertyType::biot_coefficient)
+                               .template value<double>(vars, x_position, t, dt);
+
+        // For stress dependent permeability.
+        // TODO (wenqing): get a model type, and use it to avoid the following
+        // assignment if no stress dependent permeability model is used.
+        vars[static_cast<int>(MPL::Variable::stress)].emplace<SymmetricTensor>(
+            MathLib::KelvinVector::kelvinVectorToSymmetricTensor(
+                static_cast<SymmetricTensor>(_ip_data[ip].sigma_eff -
+                                             alpha * identity2 * p_int_pt)));
```

Comment:
Why the cast?
The type of the sigma_eff*alpha*I*p is KelvinVector. Conversion to SymmetricTensor requires sqrt(2) multiplications.

Reply:
> Why the cast?

If only ``sigma_eff``, no cast is needed. `identity2` needs a cast.

> The type of the `sigma_eff*alpha*I*p` is KelvinVector. Conversion to SymmetricTensor requires sqrt(2) multiplications.

Please take a look at my new PR, #3010, about  kelvinVectorToSymmetricTensor. I assumed that function handles that multiplier correctly.

Reply:
As discussed in the closed PR #3010,  kelvinVectorToSymmetricTensor converts the Kelvin stress to conventional stress correctly.

Reply:
The `kelvinVectorToSymmetricTensor` handles the sqrt(2) multiplier correctly. static_cast is wrong way to bend the types. Use `eval`.

Reply:
use .eval().